internal/cli/cmd/dac/setup/setup.go

```diff
@@ -0,0 +1,267 @@
+	depsFolderFound := false
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+
+		// Remove the wrapping folder for following evaluations
+		currentDepPath := removeFirstFolder(header.Name)
+
+		if currentDepPath == depsFolderName {
+			depsFolderFound = true
+		}
+		if !strings.HasPrefix(currentDepPath, depsFolderName) {
+			continue
+		}
+
+		newDepPath := fmt.Sprintf("%s/%s", depsRootDstPath, currentDepPath)
+
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.Mkdir(newDepPath, 0666); err != nil {
+				return fmt.Errorf("can't create dir %s: %v", newDepPath, err)
+			}
+			logrus.Debugf("dir %s created succesfully", newDepPath)
+		case tar.TypeReg:
+			outFile, err := os.Create(newDepPath)
+			if err != nil {
+				return fmt.Errorf("can't create file %s: %v", newDepPath, err)
+			}
+			defer outFile.Close()
+			if _, err := io.CopyN(outFile, tarReader, maxFileSize); err != nil {
+				if err == io.EOF {
+					continue
+				}
+				return fmt.Errorf("can't copy content from %s: %v", header.Name, err)
+			}
+			logrus.Debugf("file %s extracted succesfully", newDepPath)
+		default:
+			return fmt.Errorf("unknown type: %b in %s", header.Typeflag, header.Name)
+		}
+	}
+
+	if !depsFolderFound {
+		return fmt.Errorf("CUE dependencies not found in archive")
+	}
+
+	return nil
+}
+
+func removeFirstFolder(filePath string) string {
+	separatorChar := "/" // force the usage of forward slash for strings comparison
+
+	// Split the path into individual components
+	components := strings.Split(filePath, separatorChar)
+
+	// Remove the top folder if there is at least one folder in the path
+	if len(components) > 1 {
+		components = components[1:]
+	}
+
+	// Join the components back into a path
+	resultPath := strings.Join(components, separatorChar)
+
+	return resultPath
+}
+
+type option struct {
+	persesCMD.Option
+	writer  io.Writer
+	version string
+}
+
+func (o *option) Complete(args []string) error {
+	if len(args) > 0 {
+		return fmt.Errorf("no args are supported by the command 'setup'")
+	}
+
+	// If no version provided, it should default to the version of the Perses server
+	if o.version == "" {
+		logrus.Debug("version flag not provided, retrieving version from Perses server..")
+		apiClient, err := config.Global.GetAPIClient()
+		if err != nil {
+			return fmt.Errorf("you need to either provide a version or be connected to a Perses server")
+		}
+
+		health, err := apiClient.V1().Health().Check()
+		if err != nil {
+			logrus.WithError(err).Debug("can't reach Perses server")
+			return fmt.Errorf("can't retrieve version from Perses server")
+		}
+		o.version = health.Version
+	}
+
+	// Add "v" prefix to the version if not present
+	if !strings.HasPrefix(o.version, "v") {
+		o.version = fmt.Sprintf("v%s", o.version)
+	}
+
+	return nil
+}
+
+func (o *option) Validate() error {
```

Comment:
what do you think about checking if the binary `cue` is accessible. I'm thinking also we could provide a flag that will give us the path to the binary `cue`. So you can point exactly what binary you want to use and also in case `cue` is not in the `PATH`

Reply:
Let's do that in a different PR. It's actually not that simple.